src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -2131,6 +2131,46 @@ private void checkInternetPermission() {
         }
     }
 
+    String serialize(final String uniqueUserId) throws AuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        }
+        
+        /* the current serialize/deserialize feature is for only supports MS apps 
+         * so the client ID for the FoCI token cache item is hard coded below
+         */        
+        final String cacheKey = CacheKey.createCacheKey(this.getAuthority(), null, AuthenticationConstants.MS_FAMILY_ID, true, uniqueUserId);
+        final TokenCacheItem tokenItem = this.getCache().getItem(cacheKey);
+        
+        if (tokenItem == null) {
+            Logger.i(TAG, "Cannot find the FoCI token cache item for this userID", "");
+            return null;
```

Comment:
I don't care too much about `null` vs `exception`, but two errors `broker present` and `FRT` not found should be exposed same way to the caller. The benefit of exception in here, that you can pass us debugging info that we can send to our telemetry and partition errors later.